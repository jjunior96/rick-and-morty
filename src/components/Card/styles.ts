import styled, { css, keyframes } from 'styled-components';
import { motion } from 'framer-motion';

const appearFromBottom = keyframes`
  from {
    opacity: 0;
    transform: translateY(20px);
  }

  to {
    opacity: 1;
    transform: translateY(0);
  }
`;

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: 360px;
    box-shadow: ${theme.box.shadow};
    border-radius: ${theme.border.radius};
    overflow: hidden;
    background-color: ${theme.colors.white};
    transition: box-shadow 0.2s;

    animation: ${appearFromBottom} 1s;

    &:hover {
      box-shadow: 0 8px 8px rgba(0, 0, 0, 0.3);
    }
  `}
`;

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 1.6rem;
  margin: 0 auto;
`;

export const Name = styled(motion.p)`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    font-weight: ${theme.font.bold};
  `}
`;

export const CardImage = styled.div`
  width: 100%;
  overflow: hidden;

  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
    transition: transform 0.4s;

    &:hover {
      transform: scale(1.2);
    }
  }
`;
