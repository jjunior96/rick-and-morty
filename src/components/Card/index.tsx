import * as S from './styles';

interface CardProps {
  name?: string;
  image?: string;
}

const Card = ({ name, image }: CardProps) => {
  return (
    <S.Container>
      {name && (
        <>
          <S.CardImage>
            <img src={image} alt={`${name} Thumbnail`} />
          </S.CardImage>
          <S.Content>
            <S.Name
              initial="hidden"
              animate="visible"
              variants={{
                hidden: {
                  scale: 0.8,
                  opacity: 0
                },
                visible: {
                  scale: 1,
                  opacity: 1,
                  transition: {
                    delay: 0.4
                  }
                }
              }}
            >
              {name}
            </S.Name>
          </S.Content>
        </>
      )}
    </S.Container>
  );
};

export default Card;
