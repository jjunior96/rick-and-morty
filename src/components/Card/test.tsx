import { screen } from '@testing-library/react';

import { renderWithTheme } from 'utils/tests/helpers';

import Card from '.';

describe('<Card />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(
      <Card title="title" img="/img/logo.svg" />
    );

    expect(
      screen.getByRole('img', { name: /thumbnail of title/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
