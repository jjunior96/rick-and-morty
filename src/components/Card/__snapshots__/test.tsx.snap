// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render the heading 1`] = `
.c0 {
  width: 100%;
  max-width: 24rem;
  height: 100%;
  max-height: 24rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  width: 100%;
  height: 100%;
  border-bottom: 4px solid #ff1015;
  overflow: hidden;
}

.c1 img {
  width: 100%;
  height: 100%;
  object-fit: cover;
  cursor: pointer;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
}

.c1 img:hover {
  -webkit-transform: scale(1.5);
  -ms-transform: scale(1.5);
  transform: scale(1.5);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 1.6rem;
}

.c2 > svg {
  cursor: pointer;
}

.c3 {
  color: #404040;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  max-width: 20rem;
  font-weight: 600;
  font-size: 2rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="Thumbnail of title"
      src="/img/logo.svg"
    />
  </div>
  <div
    class="c2"
  >
    <p
      class="c3"
    >
      title
    </p>
    <svg
      fill="currentColor"
      height="18"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 16 16"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m8 2.748-.717-.737C5.6.281 2.514.878 1.4 3.053c-.523 1.023-.641 2.5.314 4.385.92 1.815 2.834 3.989 6.286 6.357 3.452-2.368 5.365-4.542 6.286-6.357.955-1.886.838-3.362.314-4.385C13.486.878 10.4.28 8.717 2.01L8 2.748zM8 15C-7.333 4.868 3.279-3.04 7.824 1.143c.06.055.119.112.176.171a3.12 3.12 0 0 1 .176-.17C12.72-3.042 23.333 4.867 8 15z"
      />
    </svg>
  </div>
</div>
`;
